Modelica/Math/isPowerOf2.mo

```diff
@@ -3,9 +3,17 @@ function isPowerOf2 "Determine if the integer input is a power of 2"
   extends Modelica.Icons.Function;
   input Integer i(min=1) "Integer scalar";
   output Boolean result "= true, if integer scalar is a power of 2";
+protected 
+  Integer power;
+  Integer halfPower;
 algorithm
   assert(i>=1, "Integer input to isPowerOf2 has to be >= 1");
-  result := i == 2^integer(log(i)/log(2)+0.5);
+  power:=integer(log(i)/log(2)+0.5);
+  halfPower:=integer(0.5+2^div(power,2));
```

Comment:
Unrelated to what I think about the overall approach in this PR, it would be nice to have the round-to-`Integer` expressed the same way both times:
```suggestion
  power := integer(0.5 + log(i) / log(2));
  halfPower := integer(0.5 + 2^div(power, 2));
```